libs/components/forms/src/lib/modules/file-attachment/file-drop/file-drop.component.ts

```diff
@@ -432,34 +497,35 @@ export class SkyFileDropComponent implements OnDestroy {
         'skyux_file_attachment_file_upload_file_added',
         file.file.name,
       );
-    });
-
-    reader.addEventListener('error', () => {
+      this.#_uploadedFiles?.push(file);
+      this.#notifyChange?.(this.#_uploadedFiles);
+    } catch {
       fileDrop.#filesRejected(
         file,
         validFileArray,
         rejectedFileArray,
         totalFiles,
       );
-    });
-
-    reader.addEventListener('abort', () => {
-      fileDrop.#filesRejected(
-        file,
-        validFileArray,
-        rejectedFileArray,
-        totalFiles,
-      );
-    });
-
-    reader.readAsDataURL(file.file);
+    }
   }
 
-  #handleFiles(files?: FileList | null): void {
-    if (files) {
+  #handleFiles(fileList?: FileList | null | SkyFileItem[]): void {
+    if (fileList) {
       const validFileArray: SkyFileItem[] = [];
       const rejectedFileArray: SkyFileItem[] = [];
-      const totalFiles = files.length;
+      const totalFiles = fileList.length;
+
+      let files: SkyFileItem[] = [];
+
+      if ('item' in fileList) {
```

Comment:
Does this need a "truthy" check?

Reply:
if `fileList` is a FileList then its coming from the browser and item would not be undefined right?